mdp/nodes/hsfa_nodes.py

```diff
@@ -0,0 +1,540 @@
+import os
+import tempfile
+import warnings as _warnings
+import webbrowser
+import mdp
+
+
+class HSFANode(mdp.Node):
+    """Extract slowly varying components via hierarchical
+    processing from input data. The network uses weight sharing for
+    each layer over multiple locally connected receptive fields.
+
+    More information about Hierarchical Slow Feature Analysis can be found in
+    Wiskott, L. and Sejnowski, T.J., Slow Feature Analysis: Unsupervised
+    Learning of Invariances, Neural Computation, 14(4):715-770 (2002).
+
+    HSFANode also supports random sampling of data-cuboids via a RandomSwitchboard
+    instead of the fixed RectangularSwitchboard.
+
+    HSFANode also implements most of the 'list' container methods; such as insert,
+    add, delete layers from the network.
+
+    The node also provides a 'show_flow' utility method to visualize the
+    network.
+
```

Comment:
Well there are quite a few advantages: 
(a) A standardized node to implement HSFA that everyone can use. I have often heard from people complain that they get different results or sometimes terrible results. This is usually because of a few blunders in their code. 

(b) HSFANode simplifies the creation of a hsfa net without having to worry about switcboards/interconnections etc. This is especially for beginners who simply want to use HSFANode without needing to deeply understand how mdp works. The node also allows to autoset a few params by using -1.

(c) HSFANode also supports using random sampled receptive fields during training via the RandomSwitchboard. This can speed up training time on high-resolution images. 

(d) HSFANode also supports container methods like adding a layer or setting a layer with checks. This is not possible straightway with flownodes. Flownode does not include additional checks that are required to define a proper HSFANode. 

I have added a show flow just to further customize the visualization for hsfanode.